src/state_migration/common/state_migration.rs

```diff
@@ -78,18 +80,29 @@ impl<BS: Blockstore + Send + Sync> StateMigration<BS> {
         }
 
         let cache = MigrationCache::new(NonZeroUsize::new(10_000).expect("infallible"));
+        let num_threads = std::env::var("FOREST_STATE_MIGRATION_THREADS")
```

Comment:
nit: would validate that to avoid footgun.

Reply:
I forced it to at least 3.